articles/article-058-mutex.md

```diff
@@ -0,0 +1,25 @@
+<!-- BEGIN TITLE -->
+# Semaphores prevent crashes
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+Imagine a railway intersection where only one train can pass at a time. If any other train tries to pass the intersection while it's already being used, a crash happens.
+
+The same thing can happen in a computer. As soon as a file gets changed from two different sources at the same time, things can go wrong.
+
+While in rail traffic this can be solved by having a *semaphore* to only ever allow one train to pass the intersection, the same is done in computers. Here, a *semaphore* only ever allows one source to access a *critical section*. Everyone else has to wait, thus preventing any unintended behavior or crashes
+<!-- END BODY -->
```

Comment:
I would shorten the first sentence:
In rail traffic, this is solved, [...] . The same idea is used in computers.